clients/src/main/java/org/apache/kafka/clients/admin/internals/AdminApiDriver.java

```diff
@@ -250,6 +252,13 @@ public void onFailure(
                 .filter(future.lookupKeys()::contains)
                 .collect(Collectors.toSet());
             retryLookup(keysToUnmap);
+
+        } else if (t instanceof NoBatchedFindCoordinatorsException) {
```

Comment:
`AdminApiDriver` was still evolving rapidly while I was implementing this KIP so I went for the straighforward approach.

But I agree, it would be best to avoid this type of logic here. The goal would be to find a mechanism that works for all clients. @tombentley suggested an alternative option in https://github.com/apache/kafka/pull/10743#discussion_r649872433. 

I've not had the time to look into better alternatives yet.

Reply:
Should we file a Jira to not forget about improving this?

Reply:
I opened https://issues.apache.org/jira/browse/KAFKA-13013